lib/include.js

```diff
@@ -623,6 +631,66 @@ Inclusion.include = function (objects, include, options, cb) {
       }
     }
 
+
+    /**
+     * Handle Inclusion of Polymorphic HasOne relation
+     * @param callback
+     */
+    function includePolymorphicHasOne(callback) {
+      var sourceIds = [];
+      //Map for Indexing objects by their id for faster retrieval
+      var objIdMap = {};
+      for (var i = 0; i < objs.length; i++) {
+        var obj = objs[i];
+        // one-to-one: foreign key reference is modelTo -> modelFrom.
+        // use modelFrom.keyFrom in where filter later
+        var sourceId = obj[relation.keyFrom];
+        if (sourceId) {
+          sourceIds.push(sourceId);
+          objIdMap[sourceId.toString()] = obj;
+        }
+        // sourceId can be null. but cache empty data as result
+        defineCachedRelations(obj);
+        obj.__cachedRelations[relationName] = null;
+      }
+      filter.where[relation.keyTo] = {
+        inq: uniq(sourceIds)
+      };
+      relation.applyScope(null, filter);
+      relation.modelTo.find(filter, targetFetchHandler);
+      /**
+       * Process fetched related objects
+       * @param err
+       * @param {Array<Model>} targets
+       * @returns {*}
+       */
+      function targetFetchHandler(err, targets) {
+        if (err) {
+          return callback(err);
+        }
+        var tasks = [];
+        //simultaneously process subIncludes
+        if (subInclude && targets) {
+          tasks.push(function subIncludesTask(next) {
+            relation.modelTo.include(targets, subInclude, options, next);
+          });
+        }
+        //process each target object
+        tasks.push(targetLinkingTask);
+        function targetLinkingTask(next) {
+          async.each(targets, linkOneToOne, next);
+          function linkOneToOne(target, next) {
+            var sourceId = target[relation.keyTo];
+            var obj = objIdMap[sourceId.toString()];
```

Comment:
I wonder if we should check that `obj` is actually defined at this stage (defensive coding ...)? There are probably other parts of this code where this might be appropriate.